ui/src/main/webapp/src/app/analysis-context/analysis-context-form.component.ts

```diff
@@ -225,12 +230,15 @@ export class AnalysisContextFormComponent extends FormComponent
     }
 
     onSubmit() {
+        let observable = this._analysisContextService.update(this.analysisContext);
+
         // Store the Analysis Context
-        let update = this.analysisContext.id != null;
-        let service = update ? this._analysisContextService.update : this._analysisContextService.create;
-        let contextObservable: Observable<AnalysisContext> = service.call(this._analysisContextService, this.analysisContext, this.project);
+        if (this.analysisContext.id === 0 || this.analysisContext.id === null) {
```

Comment:
Yes, that is what I was referring to. I'd prefer it in either an else or just a comment. Maybe it is not confusing to someone more accustomed to frontend code than myself, but I misread it. :)